src/main/java/com/froobworld/farmcontrol/controller/action/Action.java

```diff
@@ -34,7 +35,7 @@ public boolean undoOnUnload() {
         return undoOnUnload;
     }
 
```

Comment:
Can we add a getEntityClass() method or something so the plugin knows whether an action should be applied to an entity? The entity class (e.g. Entity.class for remove and Mob.class for every other action) can be passed in the constructor

A check will need to be added to the ActionAllocationTask before any action is queued up for that entity ([around here](https://github.com/froobynooby/FarmControl/blob/07d009c520c87608d4bd6667be4993f206225bc3/src/main/java/com/froobworld/farmcontrol/controller/task/ActionAllocationTask.java#L71)) to ensure the action can actually be applied to that entity. Otherwise the action will be applied to the entity (doing nothing) but still be added to its fcData, and this will show up in the status command (e.g. minecarts would be counted as having their AI removed).

Reply:
I think I understand, we should move the check for if the action can be applied to the entity outside of the classes that implement `Action` and make some kind of util method to check if the specific action can be applied to the entity, right?

Reply:
All the other changes you made to the classes implementing `Action` are fine, I just mean we should pass in the required entity class in the constructor of `Action` as well, e.g.:
```java
public RemoveRandomMovementAction() {
    super("remove-random-movement", false, false, true, Mob.class);
}
```
and then add a getter to `Action` (like `getEntityClass()` or something) so we can add the following check to `ActionAllocationTask`:
```java
for (Action action : triggerProfilePair.actionProfile.getActions()) {
    if (!action.getEntityClass().isAssignableFrom(next.getEntityClass())) {
        continue;
    }
    TriggerActionPair triggerActionPair = new TriggerActionPair(triggerProfilePair.trigger, action);
    triggerActionPairs.add(triggerActionPair);
    unTriggerActionPairs.remove(triggerActionPair);
}
```
This will stop `ActionAllocationTask` from queuing up actions for entities that can't have that action applied to them, which will mess up their `FcData`

Reply:
Ah okay I see, I'll add this in and send the commit when I'm done. Additionally, do you think we could get in touch over Discord? I had a few other ideas I wanted to pitch/test out but I wasn't sure if they were in the scope of the project. You can add my Discord at jsinco#0

Reply:
Should be done in last commit